charts/projectsveltos/templates/deployment.yaml

```diff
@@ -435,14 +495,34 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       containers:
-{{- if .Values.agent.managementCluster }}      
-      - args: {{- toYaml .Values.classifierManager.manager.argsAgentMgmtCluster | nindent 8 -}}
-{{ else }}
-      - args: {{- toYaml .Values.classifierManager.manager.args | nindent 8 -}}
-{{- end }}
-{{- if ne (len .Values.classifierManager.agentPatchConfigMap.data) 0 }}
+    {{- if .Values.agent.managementCluster }}
+      - args:
+      {{- range .Values.classifierManager.manager.argsAgentMgmtCluster }}
+        - {{ . }}
+      {{- end }}
+      {{- range $key, $value := .Values.classifierManager.manager.extraArgsAgentMgmtCluster }}
+        {{- if $value }}
+        - --{{ $key }}={{ $value }}
+        {{- else }}
+        - --{{ $key }}
+        {{- end }}
+      {{- end }}
+    {{- else }}
+      - args:
+      {{- range .Values.classifierManager.manager.args }}
+        - {{ . }}
+      {{- end }}
+      {{- range $key, $value := .Values.classifierManager.manager.extraArgs }}
+        {{- if $value }}
+        - --{{ $key }}={{ $value }}
+        {{- else }}
+        - --{{ $key }}
+        {{- end }}
+      {{- end }}
+    {{- if ne (len .Values.classifierManager.agentPatchConfigMap.data) 0 }}
         - --sveltos-agent-config={{ .Values.classifierManager.agentPatchConfigMap.name }}
-{{- end }}
+    {{- end }}
```

Comment:
I believe this end should be at line 522.

Otherwise we are adding Values.classifierManager.agentPatchConfigMap *only* if Values.agent.managementCluster is false